google/api_core/universe_helpers.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helpers for universe domain."""
+
+from google.auth.exceptions import MutualTLSChannelError
+from typing import Any, Optional
+
+_DEFAULT_UNIVERSE = "googleapis.com"
+
+_mTLS_Universe_Error = MutualTLSChannelError(
+    f"mTLS is not supported in any universe other than {_DEFAULT_UNIVERSE}."
+)
+
+
+class _UniverseMismatchError(ValueError):
+    def __init__(self, client_universe, credentials_universe):
+        message = (
+            f"The configured universe domain ({client_universe}) does not match the universe domain "
+            f"found in the credentials ({credentials_universe}). "
+            "If you haven't configured the universe domain explicitly, "
+            f"`{_DEFAULT_UNIVERSE}` is the default."
+        )
+        super().__init__(message)
+
+
+def _get_universe_domain(
+    client_universe_domain: Optional[str], universe_domain_env: Optional[str]
+) -> str:
+    """Return the universe domain used by the client.
+
+    Args:
+        client_universe_domain (Optional[str]): The universe domain configured via the client options.
+        universe_domain_env (Optional[str]): The universe domain configured via the "GOOGLE_CLOUD_UNIVERSE_DOMAIN" environment variable.
```

Comment:
Where do we read the environment variable? Ideally, that should probably be one of the helpers too. WDYT?

(And there, potentially, we could put the name of this environment variable into a constant and possibly refer to that constant here. I like the generality of that, but since  that would be another layer of indirection for developers reading the code, maybe not.)